### PR TITLE
log2json: Detect missing RESULT lines

### LINKED ISSUE #713: [weekly reports] display total number of tests to be run in summary
We are gathering also results of nightly tests that timeout, ie not the whole batch is run.
For example the last three daily-iso runs of https://github.com/rhinstaller/kickstart-tests/actions/runs/2218861575
One of them has only 217 tests run by the timeout:
```
04/24/2022
    logs-daily-iso (Ran 217, 216 passed, 1 failed. 3 flakes) :
```
While the test from the beginning of the week run all 220 tests:
```
04/19/2022
    logs-daily-iso (Ran 220, 220 passed, 0 failed. 3 flakes) :
```
It is not easy to notice that for the former all the tests were not actually run. Maybe we could add something like `Total` (and perhaps unify the string formatting a bit):
```
04/24/2022
    logs-daily-iso (Total 220, ran 217, 216 passed, 1 failed. 3 flakes) :
```
We don't know the number of test before running the test, I used to parse it out (split and count) from this line of `kstest.log`:
```
2022-04-24T23:07:55.3823805Z Running tests: ./bridge-httpks.sh ./packages-instlangs-1.sh ./bindtomac-bridge-httpks.sh ./vlan-pre.sh ./ui_vnc.sh ./initial-setup-default.sh ./team-pre.sh ./network-bootopts-bond-dhcp-httpks.sh ./ntp-nontp-without-chrony-gui.sh ./dns.sh ./bootloader-3.sh ./reboot-initial-setup-gui.sh ./network-addr-gen-mode-dhcpall.sh ./bindtomac-onboot-bootopts-pre.sh ./bindtomac-ifname-httpks.sh ./network-static-to-dhcp-pre-single.sh ./network-device-mac-httpks.sh ./proxy-cmdline.sh ./driverdisk-disk.sh ./keyboard-convert-x-override-bootopt.sh ./team-httpks.sh ./packages-and-groups-ignoremissing.sh ./team.sh ./bond-vlan-pre.sh ./mountpoint-assignment-1.sh ./packages-and-groups-1.sh ./bindtomac-bridge-2devs-httpks.sh ./reqpart.sh ./module-3.sh ./lvm-1.sh ./geolocation-off-by-default-with-ks.sh ./default-systemd-target-tui.sh ./ui_graphical_interactive.sh ./groups-and-envs-2.sh ./nosave-3.sh ./default-desktop.sh ./network-bootopts-static-unspec-single.sh ./bindtomac-network-static-to-dhcp-pre-single.sh ./packages-ignorebroken.sh ./network-noipv4-pre.sh ./ntp-pools.sh ./rpm-ostree.sh ./initial-setup-enable.sh ./repo-addrepo.sh ./clearpart-1.sh ./services.sh ./dracut-visible-warnings.sh ./bootloader-1.sh ./module-1.sh ./network-bootopts-bond-ks-override.sh ./reboot.sh ./onboot-activate.sh ./bindtomac-bridge-2devs-pre.sh ./https-repo.sh ./bond2-httpks.sh ./hostname-bootopts.sh ./bridged-bond-httpks.sh ./lang.sh ./user-locked-root-locked-admin.sh ./repo-install.sh ./tmpfs-fixed_size.sh ./ui_text_noninteractive.sh ./anaconda-modules.sh ./url-mirrorlist.sh ./bond2-pre.sh ./firewall-disable-with-options.sh ./bootloader-5.sh ./keyboard.sh ./autopart-nohome.sh ./packages-instlangs-3.sh ./network-device-default-ksdevice-pre.sh ./ntp-without-chrony-gui.sh ./packages-ignoremissing.sh ./bindtomac-bond2-pre.sh ./bond-vlan-httpks.sh ./selinux-disabled.sh ./module-ignoremissing.sh ./bindtomac-bond2-httpks.sh ./ui_text_interactive.sh ./network-options-pre.sh ./bridge-2devs.sh ./network-device-default-pre-hostname.sh ./bond.sh ./bridge-2devs-httpks.sh ./authselect.sh ./default-systemd-target-gui-graphical-provides.sh ./hello-world.sh ./reboot-initial-setup-tui.sh ./default-systemd-target-startxonboot.sh ./url-metalink.sh ./repo-exclude.sh ./network-device-mac-pre.sh ./network-static-2-pre.sh ./lvm-2.sh ./autopart-encrypted-2.sh ./default-fstype.sh ./firewall-use-system-defaults.sh ./network-bootopts-team-dhcp-httpks.sh ./network-device-default.sh ./network-bootopts-static-mac.sh ./encrypt-device.sh ./nosave-1.sh ./network-bootopts-static.sh ./default-systemd-target-vnc.sh ./onboot-bootopts-pre.sh ./autopart-luks-3.sh ./network-bootopts-noautodefault.sh ./default-systemd-target-vnc-graphical-provides.sh ./harddrive-install-tree.sh ./rootpw-crypted.sh ./user-multiple.sh ./network-bootopts-bootif-httpks.sh ./repo-metalink.sh ./bindtomac-network-device-mac.sh ./snapshot-pre.sh ./firewall-use-system-defaults-ignore-options.sh ./encrypt-swap.sh ./pre-install.sh ./rootpw-basic.sh ./ntp-with-nontp-gui.sh ./packages-excludedocs.sh ./bindtomac-bond-vlan-httpks.sh ./bootloader-4.sh ./bindtomac-network-device-mac-httpks.sh ./firewall.sh ./ifname-httpks.sh ./groups-ignoremissing.sh ./selinux-contexts.sh ./hostname.sh ./ui_cmdline.sh ./autopart-luks-2.sh ./autopart-luks-5.sh ./harddrive-install-tree-relative.sh ./bridge-2devs-pre.sh ./firewall-disable.sh ./network-missing-ifcfg-httpks.sh ./clearpart-3.sh ./bindtomac-bridged-bond-httpks.sh ./network-addr-gen-mode.sh ./bootloader-2.sh ./network-device-mac.sh ./keyboard-convert-vc.sh ./bridge-no-bootopts-net.sh ./network-device-bootif-httpks.sh ./packages-instlangs-2.sh ./driverdisk-disk-kargs.sh ./default-systemd-target-gui.sh ./bindtomac-network-static-2-httpks.sh ./timezoneLOCAL.sh ./image-deployment-2.sh ./bindtomac-network-device-mac-pre.sh ./bindtomac-team-pre.sh ./module-4.sh ./fedora-live-image-build.sh ./network-autoconnections-httpks.sh ./rootpw-lock-no-password.sh ./keyboard-bootopt-only.sh ./network-noipv4-httpks.sh ./network-bootopts-vlan-static-httpks.sh ./user-wheel-no-root.sh ./network-bootopts-bridge-dhcp-httpks.sh ./ntp-nontp-without-chrony.sh ./lvm-thinp-1.sh ./repo-enable.sh ./network-static-httpks.sh ./network-bootopts-static-unspec-bootif.sh ./initial-setup-reconfig.sh ./autopart-fstype.sh ./keyboard-generic-argument.sh ./btrfs-2.sh ./default-systemd-target-tui-graphical-provides.sh ./network-device-default-ksdevice-httpks.sh ./repo-baseurl.sh ./autopart-encrypted-1.sh ./onboot-activate-httpks.sh ./nosave-2.sh ./container.sh ./autopart-encrypted-3.sh ./rootpw-lock.sh ./url-baseurl.sh ./bindtomac-onboot-activate-httpks.sh ./repo-mirrorlist.sh ./escrow-cert.sh ./groups-and-envs-1.sh ./repo-include.sh ./anaconda-conf.sh ./autopart-luks-1.sh ./network-static.sh ./packages-default.sh ./network-static-2-httpks.sh ./bindtomac-network-device-default-httpks.sh ./ntp-without-chrony.sh ./packages-weakdeps.sh ./selinux-permissive.sh ./bindtomac-team-httpks.sh ./autopart-luks-4.sh ./bridged-bond-pre.sh ./btrfs-1.sh ./clearpart-2.sh ./vlan-httpks.sh ./user-single.sh ./ignoredisk-1.sh ./default-systemd-target-skipx.sh ./groups-and-envs-3.sh ./initial-setup-gui.sh ./network-static-to-dhcp-pre.sh ./user-multiple-wheel-no-root.sh ./ks-include.sh ./ntp-with-nontp.sh ./network-autoconnections-dhcpall-httpks.sh ./timezoneUTC.sh ./selinux-enforcing.sh ./ignoredisk-2.sh ./raid-1.sh ./initial-setup-disable.sh ./mountpoint-assignment-2.sh ./network-device-default-httpks.sh ./lvm-thinp-2.sh ./lvm-raid-1.sh ./module-2.sh 
```

### PR DESCRIPTION
**Pull request by bcl**
Some tests can fail without emitting a RESULT line. Catch these cases
and create a fake RESULT entry, and guess at the test name based on the
install_log filename. If the format of the kstest-... part of that
filename ever changes this will need to be updated.

Fixes #713

### CODE REVIEW

**Review by VladimirSlavik — APPROVED** [review]
Looks good to me. Thank you!

**Comment by rvykydal** [comment]
Looks good to me, but I think it would be useful to add a "missing" category to the summary (perhaps in a separate follow-up PR if you prefer).
I've run the summary on the https://github.com/rvykydal/kickstart-tests/actions/runs/2232478365 - attaching the full output:

[missing-tests.log](https://github.com/rhinstaller/kickstart-tests/files/8619104/missing-tests.log)

The typical case where we hit missing tests is when the test batch exceeds the timeout limit and is terminated, leaving
```
initial-setup-enable:
        04/27/2022 11:10:51 (0s): Missing RESULT line
``` 
for the tests that already started but were killed,
and
```
default-systemd-target-skipx:
         (0s): Missing test
```
for the tests that haven't started.

It is useful to look at real failures of the tests that finished even in this case (incomplete batch run), so to have the missing results distinct from real failures. Which could be improved by
1) adding "missing" category to the summary:
instead of:
```
logs-daily-iso: Ran 220 tests. 30 passed, 190 failed, 0 flakes.
...
04/27/2022
    logs-daily-iso (Ran 220, 30 passed, 190 failed. 0 flakes) :
```
have:
```
logs-daily-iso: Ran 220 tests. 30 passed, 0 failed, 190 missing, 0 flakes.
...
04/27/2022
    logs-daily-iso (Ran 220, 30 passed, 0 failed, 190 missing, 0 flakes) :
```
2) Adding missing to daily summary
```
04/27/2022
    logs-daily-iso (Ran 220, 30 passed, 0 failed, 190 missing. 0 flakes) :
        Missing:
            anaconda-conf
            anaconda-modules
            authselect
            autopart-encrypted-1

```
3) Not showing details of missing tests.

4) Also the missing tests included in failed are polluting information (I found quite nice and useful) provided by the Top 5 failures.

**Comment by bcl** [comment]
That makes sense, I'll do that as part of this PR.

**Review by VladimirSlavik — APPROVED** [review]
Thank you, still LGTM.

File: scripts/weekly-summary [hunk 1]
```diff
@@ -316,19 +321,25 @@ def archive_test_logs(days, archive_path, all_logs):
...
                 for t in sorted(flakes[name], key=lambda x: x["start_time"]):
-                    logdir = kstest_logdir(tmpdir, t)
-                    if not os.path.exists(logdir):
-                        raise RuntimeError(f"Missing logdir - {logdir}")
+                    try:
+                        logdir = kstest_logdir(tmpdir, t)
+                        if not logdir or not os.path.exists(logdir):
+                            raise RuntimeError(f"Missing logdir - {logdir}")
+                    except RuntimeError:
+                        continue
```
[the remark this answers is not in the document]
Comment (bcl) [reply scripts/weekly-summary:372]: kstest_logdir can also raise RuntimeException so it needs to be wrapped if you want to skip it and continue.

File: scripts/weekly-summary [hunk 2]
```diff
@@ -316,19 +321,25 @@ def archive_test_logs(days, archive_path, all_logs):
...
                 for t in sorted(failed[name], key=lambda x: x["start_time"]):
-                    logdir = kstest_logdir(tmpdir, t)
-                    if not os.path.exists(logdir):
-                        raise RuntimeError(f"Missing logdir - {logdir}")
+                    try:
+                        logdir = kstest_logdir(tmpdir, t)
+                        if not os.path.exists(logdir):
+                            raise RuntimeError(f"Missing logdir - {logdir}")
+                    except RuntimeError:
+                        continue
```
[the remark this answers is not in the document]
Comment (bcl) [reply scripts/weekly-summary:359]: lol. Good question :) I think I was debating adding logging to the except path but didn't make up my mind.

**Comment by bcl** [comment]
@rvykydal can you merge this?

**Pull request merged**